src/Model.php

```diff
@@ -1220,7 +1212,7 @@ public function asModel(string $class, array $options = []): self
         $m = $this->newInstance($class, $options);
 
         foreach ($this->data as $field => $value) {
-            if ($value !== null && $value !== $this->getField($field)->default) {
+            if ($value !== null && $value !== $this->getField($field)->default && $field !== $this->id_field) {
```

Comment:
Why dont you want id_field copied? I think we need to define what asModel() does:
1) Should it work like this (I don't like the code):
```
$person = (new SomePerson($persistence))->load(44);
$admin = $person->asModel(Admin::class);
$admin->set('something', 'someValue'');
$admin->save(); //Will overwrite record with ID 44
```

2) Or should it create a copy of the record like:
```
$person = (new SomePerson($persistence))->load(44);
$admin = $person->asModel(Admin::class);
$admin->set('something', 'someValue'');
$admin->save(); //Will add new record to Persistence
```

Personally, I don't see much use for 1), so for me code change is ok. What do others think?

Reply:
ID is normally not globally unique and it is assigned usually by on insert "by the target table", thus we do not copy ID by default

Reply:
Ah, I get your thought. I did not think about the use case that the models are saved into different tables. You are right, very sensible!